### PR TITLE
Make zc_nvlist_src_size limit tunable

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

We limit the size of nvlists passed to the kernel so a user cannot make
the kernel do an unreasonably large allocation.  On FreeBSD this limit
was 128 kiB, which turns out to be a bit too small when doing some
operations involving a large number of datasets or snapshots, for
example replication.

### Description

Make this limit tunable, with a platform-specific auto default.
Linux keeps its limit at KMALLOC_MAX_SIZE. FreeBSD uses 1/4 of the
system limit on user wired memory, which allows it to scale depending
on system configuration.

### How Has This Been Tested?

Made a dataset with 10,000 snapshots (way too many, in hindsight).
Send fails before the change, succeeds after.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
I believe this is the issue may be related #6572

**Comment by freqlabs** [comment]
`uint64_t` -> `unsigned long` to fix Linux build

File: module/zfs/zfs_ioctl.c [hunk 1]
```diff
@@ -7399,7 +7470,8 @@ zfsdev_ioctl_common(uint_t vecnum, zfs_cmd_t *zc, int flag)
 		return (SET_ERROR(ZFS_ERR_IOC_CMD_UNAVAIL));
 
 	zc->zc_iflags = flag & FKIOCTL;
-	if (zc->zc_nvlist_src_size > MAX_NVLIST_SRC_SIZE) {
```
Comment (behlendorf) [inline module/zfs/zfs_ioctl.c:7402]: `MAX_NVLIST_SRC_SIZE` is now unused and the define can be removed.

Comment (freqlabs) [reply module/zfs/zfs_ioctl.c:7402]: Yes it was removed.

Comment (freqlabs) [reply module/zfs/zfs_ioctl.c:7402]: I might be able to remove the FreeBSD definition of `KMALLOC_MAX_SIZE` too, though. I'll test that out.

**Comment by freqlabs** [comment]
* Remove FreeBSD's definition of `KMALLOC_MAX_SIZE`
* Add `vm_page_max_user_wired` compat shim for FreeBSD 12.1-RELEASE

File: module/zfs/zfs_ioctl.c [hunk 2]
```diff
@@ -6822,188 +6825,215 @@ zfs_ioctl_register(const char *name, zfs_ioc_t ioc, zfs_ioc_func_t *func,
 }
 
 static void
-zfs_ioctl_register_pool(zfs_ioc_t ioc, zfs_ioc_legacy_func_t *func,
-    zfs_secpolicy_func_t *secpolicy, boolean_t log_history,
-    zfs_ioc_poolcheck_t pool_check)
+zfs_ioctl_register_pool(const char *name, zfs_ioc_t ioc,
+    zfs_ioc_legacy_func_t *func, zfs_secpolicy_func_t *secpolicy,
+    boolean_t log_history, zfs_ioc_poolcheck_t pool_check)
```
Comment (behlendorf) [inline module/zfs/zfs_ioctl.c:6830]: I'd prefer not to make these zfs_ioctl changes in this PR for the sake of a slightly improved `zfs_dbgmsg()`.  I'm not against them per-say, but they obscure the issue being addressed here.  If we're interested in making the ioctl change I'd suggest moving them to their own PR and we can solicit additional feedback.

**Comment by freqlabs** [comment]
* Only the changes for the size limit tunable itself
